clap-v3-utils/src/keypair.rs

```diff
@@ -1020,6 +1021,103 @@ fn confirm_keypair_pubkey(keypair: &Keypair) {
     }
 }
 
+/// Loads an [ElGamalKeypair] from one of several possible sources.
+///
+/// If `confirm_pubkey` is `true` then after deriving the keypair, the user will
+/// be prompted to confirm that the ElGamal pubkey is as expected.
+///
+/// The way this function interprets its arguments is analogous to that of
+/// [`signer_from_path`].
+///
+/// The bip32 hierarchical derivation of an ElGamal keypair is not currently
+/// supported.
+///
+/// # Examples
+///
+/// ```no_run`
+/// use clap::{Arg, Command};
+/// use solana_clap_v3_utils::keypair::elgamal_keypair_from_path;
+///
+/// let clap_app = Command::new("my-program")
+///     // The argument we'll parse as a signer "path"
+///     .arg(Arg::new("elgamal-keypair")
+///         .required(true)
+///         .help("The default signer"));
+///
+/// let clap_matches = clap_app.get_matches();
+/// let elgamal_keypair_str: String = clap_matches.value_of_t_or_exit("elgamal-keypair");
+///
+/// let elgamal_keypair = elgamal_keypair_from_path(
+///     &clap_matches,
+///     &elgamal_keypair_str,
+///     "elgamal-keypair",
+///     false,
+/// )?;
+/// # Ok::<(), Box<dyn std::error::Error>>(())
+/// ```
+pub fn elgamal_keypair_from_path(
+    matches: &ArgMatches,
+    path: &str,
+    elgamal_keypair_name: &str,
+    confirm_pubkey: bool,
+) -> Result<ElGamalKeypair, Box<dyn error::Error>> {
+    let elgamal_keypair = encodable_key_from_path(matches, path, elgamal_keypair_name)?;
+    if confirm_pubkey {
+        confirm_elgamal_keypair_pubkey(&elgamal_keypair);
+    }
+    Ok(elgamal_keypair)
+}
+
+fn confirm_elgamal_keypair_pubkey(keypair: &ElGamalKeypair) {
+    let elgamal_pubkey = keypair.public;
```

Comment:
No, it doesn't really need to be public... This was something that I was thinking about changing as well, but lost out on other changes especially since I was afraid that it will break downstream projects. I will also create a PR for this as well...